packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -812,6 +814,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
         target={ this._comboBoxWrapper }
         onDismiss={ this._onDismiss }
         setInitialFocus={ false }
+        calloutWidth={ dropdownWidth || this._comboBoxWrapper.clientWidth + 2 }
```

Comment:
The problem with this approach is that it will introduce another reflow from the measurement to get the clientWidth. Ideally, we'd only do a measurement once (inside the callout component) Where does the +2 come from?

Reply:
Though I guess there was already a reflow from line 255

Reply:
@christiango, _comboBoxWrapper wil have its reference set after combobox loads. Without clicking the dropdown,  the callout is not rendered. Then when user click on the dropdown button, callout will try to render, and at this point, the ref has already been set, and it'll flow once to callout. Am I missing anything? The 2px is the reverse engineer on the setWidth setWidth = isOpen ? (this._comboBoxWrapper.clientWidth - 2) + 'px' : undefined; Otherwise, the width wouldn't cover the whole width of the target element.

Reply:
So callout rendereing is already pretty slow. Since the callout is using _comboBoxWrapper as a target, it's bounds will be computed by the callout (forcing a reflow). The call to this_comboBoxWrapper.clientWidth seems like it would add a second reflow and make the frame for opening a callout even longer. This could be verified by looking at a perf trace in the browser dev tools before and after this change.

Reply:
Thanks for pointing out the reflow @christiango , I took a look at the calloutContent, which is the piece that has calculation and cause reflow. The way they are using calloutWidth is passing in through mergeStyles and use it as className on the container wrapping up the popup. 
 className={
            mergeStyles(
              'ms-Callout',
              styles.root,
              className,
              directionalClassName,
              !!calloutWidth && { width: calloutWidth }
            ) }
So I don't think by passing that prop in, it'll impacting the existing reflow paths. I also recorded the render time and methods of calloutContent
Render time without passing calloutWidth
![image](https://user-images.githubusercontent.com/8844898/33103407-0d4c21d2-ced7-11e7-90c5-80fb4a41d6db.png)
Render time with passing calloutWidth
![image](https://user-images.githubusercontent.com/8844898/33103410-0f600506-ced7-11e7-8ff4-521bc62a8fc1.png)

Reply:
Thanks for pointing out the reflow @christiango , I took a look at the calloutContent, which is the piece that has calculation and cause reflow. The way they are using calloutWidth is passing in through mergeStyles and use it as className on the container wrapping up the popup.
className={
mergeStyles(
'ms-Callout',
styles.root,
className,
directionalClassName,
!!calloutWidth && { width: calloutWidth }
) }
So I don't think by passing that prop in, it'll impacting the existing reflow paths. I also recorded the render time and methods of calloutContent
Render time without passing calloutWidth and path
![image](https://user-images.githubusercontent.com/8844898/33104458-21af69c6-cedd-11e7-83fc-f90d977c06f9.png)
Render time with passing calloutWidth and path
![image](https://user-images.githubusercontent.com/8844898/33104467-2a76ef5c-cedd-11e7-84f4-8d5e92d63d2c.png)

Reply:
Thanks for looking into it! I would expect the code you are adding to add a new reflow higher up in the callstack since I expect the the _comboBoxWrapper.clientWidth call would take place before the callout content gets rendered.